File: android/app/src/main/assets/debug/env/v8/page_url.js

class URL {
    constructor(url) {
        let str = url;
        let potocal_reg = str.match(/^\w+:/);
        if (!potocal_reg) {
            return;
        } 
        this.protocol = potocal_reg[0];
        str = str.substr(this.protocol.length + 2);
        this.host = this.hostname = str.substr(0, str.indexOf('/'));
        str = str.substr(this.host.length);
        let path = str;
        let sidx = path.indexOf('?');
        let search;
        if (sidx >= 0) {
            this.pathname = this.path = path.substr(0, sidx);
            search = path.substr(sidx);
        } else {
            this.pathname = this.path = path;
            search = '';
        }
        if (!this.path) this.path = "/";
        let hidx = search.indexOf('#');
        if (hidx >= 0) {
            this.search = search.substr(0, hidx);
            this.hash = search.substr(hidx);
        } else {
            this.search = search;
            this.hash = '';
        }
        this.origin = this.protocol + '//' + this.hostname;
        this.href = this.origin + this.path + this.search + this.hash;

        let query = {};
        this.search.substr(1).split('&').forEach(function(str) {
            let idx = str.indexOf('=');
            if (idx > 0) {
                query[str.substr(0, idx)] = str.substr(idx + 1);
            }
        });
        this.searchParams = query;
    }
}

class PageURL {
    constructor(url) {
        this.url = new URL(url);
    }

    href(src) {
        if (src.match(/^\w+:/)) {
            return src;
        } else if (src[0] === '/') {
            return this.url.origin + src;
        } else {
            let bsegs = this.url.pathname.split('/');
            let segs = src.split('/');
            for (let seg of segs) {
                if (seg === '.') {

                } else if (seg === '..') {
                    bsegs.pop();
                } else {
                    bsegs.push(seg);
                }
            }
            let path = bsegs.join('/');
            if (path[0] !== '/') {
                path = '/' + path;
            }
            return this.url.origin + path;
        }
    }
}

module.exports = {
    PageURL,
    URL
};

File: android/app/src/main/assets/debug/env/ruby/page_url.rb
module URI
  class Parsed
    def hostname
      host
    end
  end
end

class PageURL
  def initialize url
    p "init #{url}"
    @uri = URI.parse url
  end

  def origin
    str = "#{@uri.scheme}://#{@uri.hostname}"
  end

  def href src
    if src.match(/^\w+:/)
      src
    elsif src[0] == '/'
      origin + src
    else
      bsegs = @uri.path.split '/'
      segs = src.split '/'
      segs.each do |seg|
        if seg == '.'
        elsif seg == '..'
          bsegs.pop
        else
          bsegs << seg
        end
      end
      path = bsegs * '/'
      path = "/#{path}" unless path[0] == '/'
      origin + path
    end
  end
end